optuna/samplers/_nsgaiii/_elite_population_selection_strategy.py

```diff
@@ -102,55 +101,30 @@ def __call__(self, study: Study, population: list[FrozenTrial]) -> list[FrozenTr
         return elite_population
 
 
-def _multi_choose(n: int, k: int) -> int:
-    return math.comb(n + k - 1, k)
-
-
 def _generate_default_reference_point(
     n_objectives: int, dividing_parameter: int = 3
 ) -> np.ndarray:
     """Generates default reference points which are `uniformly` spread on a hyperplane."""
-    reference_points = np.zeros(
-        (
-            _multi_choose(n_objectives, dividing_parameter),
-            n_objectives,
-        )
+    indices = np.array(
+        list(combinations_with_replacement(range(n_objectives), dividing_parameter))
     )
-    for i, comb in enumerate(
-        itertools.combinations_with_replacement(range(n_objectives), dividing_parameter)
-    ):
-        for j in comb:
-            reference_points[i, j] += 1.0
+    row_indices = np.repeat(np.arange(len(indices)), dividing_parameter)
+    col_indices = indices.flatten()
+    reference_points = np.zeros((len(indices), n_objectives), dtype=float)
+    np.add.at(reference_points, (row_indices, col_indices), 1.0)
     return reference_points
 
 
 def _filter_inf(population: list[FrozenTrial]) -> np.ndarray:
-    # Collect all objective values.
-    n_objectives = len(population[0].values)
-    objective_matrix = np.zeros((len(population), n_objectives))
+    objective_matrix = np.zeros((len(population), len(population[0].values)))
```

Comment:
I am not sure if I can assume that `population[0].values` is not None, but I followed the original code.

Reply:
I confirmed that population is a list of completed trials.
- [generation_to_population only includes completed trials](https://github.com/optuna/optuna/blob/master/optuna/samplers/_nsgaiii/_sampler.py#L219-L220)
- [population is created from `generation_to_population`](https://github.com/optuna/optuna/blob/master/optuna/samplers/_nsgaiii/_sampler.py#L227)